### PR TITLE
Fix(tsql): preserve column projection quotes for newly added Alias nodes

### PR DESCRIPTION
**Pull request by georgesittas**

### CODE REVIEW

File: sqlglot/dialects/tsql.py [hunk 1]
```diff
@@ -213,7 +213,25 @@ def qualify_derived_table_outputs(expression: exp.Expression) -> exp.Expression:
...
+            if isinstance(c, exp.Column) and not c.alias
+        )
+
+        qualify_outputs(subqueryable)
+
+        # Preserve the quoting information of columns for newly added Alias nodes
+        subqueryable_selects = subqueryable.selects
+        for select_index in unaliased_column_projection_indexes:
+            select = subqueryable_selects[select_index]
+            if isinstance(select, exp.Alias):
```
Comment (tobymao) [inline sqlglot/dialects/tsql.py:232]: shouldn't this always be an alias because we qualify outputs?

Comment (georgesittas) [reply sqlglot/dialects/tsql.py:232]: yes it's a sanity check, I can probably get rid of it

File: sqlglot/dialects/tsql.py [hunk 2]
```diff
@@ -213,7 +213,25 @@ def qualify_derived_table_outputs(expression: exp.Expression) -> exp.Expression:
...
+                column_identifier = select.this.this
+                select.args["alias"].set("quoted", column_identifier.args.get("quoted"))
```
Comment (tobymao) [inline sqlglot/dialects/tsql.py:234]: maybe let's check that this is an idintifier and called identifier.quoted

File: sqlglot/dialects/tsql.py (hunk 1 above)
Comment (georgesittas) [reply sqlglot/dialects/tsql.py]: removed

File: sqlglot/dialects/tsql.py (hunk 2 above)
Comment (georgesittas) [reply sqlglot/dialects/tsql.py]: added

File: sqlglot/dialects/tsql.py [hunk 3]
```diff
@@ -213,7 +213,25 @@ def qualify_derived_table_outputs(expression: exp.Expression) -> exp.Expression:
...
+            if isinstance(c, exp.Column) and not c.alias
+        )
+
+        qualify_outputs(subqueryable)
+
+        # Preserve the quoting information of columns for newly added Alias nodes
+        subqueryable_selects = subqueryable.selects
+        for select_index in unaliased_column_indexes:
+            alias = subqueryable_selects[select_index]
+            column_identifier = alias.this.this
```
Comment (tobymao) [inline sqlglot/dialects/tsql.py:232]: this should be alias.this, and check that alias.this is a column

Comment (georgesittas) [reply sqlglot/dialects/tsql.py]: yea makes sense, changed

Comment (georgesittas) [reply sqlglot/dialects/tsql.py]: > and check that alias.this is a column

I assume you meant "is an identifier" instead?

**Pull request merged**